### PR TITLE
Run the unimportantLoopCount at all optlevels

### PR DESCRIPTION
**Pull request by JamesKingdon**
In TR_LoopVersioner::loopIsWorthVersioning we currently only
run the test for unimportantLoopCount for optlevels <= warm. It is
likely to be beneficial in many circumstances to run this test at
all optlevels.

This change is currently guarded by the Xjit option added in
PR #1369 until such time as the behaviour is tested accross a broad
set of applications.

Signed-off-by: JamesKingdon <jkingdon@ca.ibm.com>

### CODE REVIEW

File: compiler/optimizer/LoopVersioner.cpp [hunk 1]
```diff
@@ -187,7 +187,9 @@ bool TR_LoopVersioner::loopIsWorthVersioning(TR_RegionStructure *naturalLoop)
    if (entryBlock->isCold())
       return false;
 
-   if (comp()->getMethodHotness() <= warm)
+   // if aggressive loop versioner flag is set then run at any optlevel, otherwise only at warm or less
+   bool aggressive = TR::Options::getCmdLineOptions()->getOption(TR_EnableAggressiveLoopVersioning);
+   if (aggressive || comp()->getMethodHotness() <= warm)
```
Comment (andrewcraik) [inline compiler/optimizer/LoopVersioner.cpp]: This probably deserves a trace message to show the decision made - it should be guarded for tracing of the opt in question.

**Comment by JamesKingdon** [comment]
@andrewcraik Additional trace added.

**Comment by vijaysun-omr** [comment]
Change looks okay to me. Could you please re-base and ensure checks pass so that I can consider merging ?

**Comment by vijaysun-omr** [comment]
Checks have passed. Merging.

**Pull request merged**